fhir-search/src/test/java/com/ibm/fhir/search/reference/ReferenceTypeTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.search.reference;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.search.parameters.QueryParameterValue;
+import com.ibm.fhir.search.reference.value.CanonicalUrlHandlerImpl;
+import com.ibm.fhir.search.reference.value.IdHandlerImpl;
+import com.ibm.fhir.search.reference.value.TypeIdHandlerImpl;
+import com.ibm.fhir.search.reference.value.UrlHandlerImpl;
+
+public class ReferenceTypeTest {
```

Comment:
what does it do for parameters that can point at multiple different types?

Reply:
The code matrixes the targets creating multiple Search Parameter Values